src/transformers/models/sam/modeling_sam.py

```diff
@@ -576,8 +576,12 @@ def __init__(self, config: SamPromptEncoderConfig):
         self.conv1 = nn.Conv2d(1, self.mask_input_channels, kernel_size=2, stride=2)
         self.conv2 = nn.Conv2d(self.mask_input_channels, config.mask_input_channels, kernel_size=2, stride=2)
         self.conv3 = nn.Conv2d(config.mask_input_channels, config.hidden_size, kernel_size=1)
-        self.layer_norm1 = SamLayerNorm(self.mask_input_channels, config.layer_norm_eps)
-        self.layer_norm2 = SamLayerNorm(self.mask_input_channels * 4, config.layer_norm_eps)
+        self.layer_norm1 = SamLayerNorm(
+            self.mask_input_channels, eps=config.layer_norm_eps, data_format="channels_first"
+        )
+        self.layer_norm2 = SamLayerNorm(
+            self.mask_input_channels * 4, eps=config.layer_norm_eps, data_format="channels_first"
+        )
```

Comment:
@younesbelkada can you confirm if this a mistake in the SAM implementation?

Reply:
Yes I had to include this fix to make `input_masks` work. I noticed that `input_masks` is not tested in `tests/models/test_modeling_sam.py` (only `input_points`, `input_labels` and `input_boxes` are).

Might be good to add a test for this

Reply:
Sounds good to me! Indeed we didn't tested input masks!